core/src/server/rpc_module.rs

```diff
@@ -806,27 +885,41 @@ impl SubscriptionSink {
 					Either::Left((Ok(Some(result)), next_closed_fut)) => {
 						match self.send(&result) {
 							Ok(_) => (),
-							Err(Error::SubscriptionClosed(close_reason)) => {
-								self.close(&close_reason);
+							Err(Error::SubscriptionClosed(close)) => {
+								self.close(&ErrorObject::code_and_message(
+									SUBSCRIPTION_CLOSED_WITH_ERROR,
+									&close.to_string(),
+								));
 								break Ok(());
 							}
 							Err(err) => {
-								break Err(err);
+								let reason = err.to_string();
+								self.close(&ErrorObject::code_and_message(SUBSCRIPTION_CLOSED_WITH_ERROR, &reason));
```

Comment:
There is no way to opt from these close messages but I argued that one could use `manual send` if that breaks things.

However, we could make this `&mut self` and let the user close it manually if it wants it because this might break compat with some libraries works with polkadot JS at least.